File: frontend/src/modules/codeEditor/services/codeExecutionService.ts
import axios from 'axios';

const API_URL = import.meta.env.VITE_API_URL || 'http://localhost:5000';

export interface CodeExecutionResult {
  status: {
    id: number;
    description: string;
  };
  stdout: string;
  stderr: string;
  memory: string;
  time: string;
  input_used?: number;
}

export const executeCode = async (code: string, input?: string[]): Promise<CodeExecutionResult> => {
  try {
    console.log('Enviando código al backend:', code);
    console.log('Input proporcionado:', input);
    
    const payload: any = { code };
    if (input && input.length > 0) {
      payload.input = input;
    }
    
    const response = await axios.post(`${API_URL}/api/code/execute`, payload);
    console.log('Respuesta del backend:', response);
    return response.data;
  } catch (error) {
    console.error('Error al ejecutar el código:', error);
    if (axios.isAxiosError(error)) {
      const errorMessage = error.response?.data?.error || error.message || 'Error al ejecutar el código';
      throw new Error(errorMessage);
    }
    throw error;
  }
}; 

File: backend/app/services/code_execution_service.py
import requests
from flask import current_app
from typing import Dict, Any

class CodeExecutionService:
    def __init__(self):
        self.sandbox_url = current_app.config.get('CODE_SANDBOX_URL', 'http://code-sandbox:5001')

    def execute_code(self, code: str, input_data: str = '') -> Dict[str, Any]:
        """
        Ejecuta código Python usando el sandbox.
        
        Args:
            code (str): Código Python a ejecutar
            input_data (str): Datos de entrada para stdin (opcional)
            
        Returns:
            Dict[str, Any]: Resultado de la ejecución con el siguiente formato:
            {
                'status': {'id': int, 'description': str},
                'stdout': str,
                'stderr': str,
                'memory': str,
                'time': str
            }
        """
        try:
            response = requests.post(
                f"{self.sandbox_url}/execute",
                json={"code": code, "input": input_data},
                timeout=10
            )
            response.raise_for_status()
            return response.json()
        except requests.exceptions.RequestException as e:
            current_app.logger.error(f"Error al comunicarse con el sandbox: {str(e)}")
            return {
                'status': {'id': 4, 'description': 'Error'},
                'stdout': '',
                'stderr': f"Error al comunicarse con el sandbox: {str(e)}",
                'memory': '0',
                'time': '0'
            } 